plugins/wporg-5ftf/views/form-my-pledges.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace WordPressDotOrg\FiveForTheFuture\View;
+use WP_User, WP_Post;
+
+/**
+ * @var WP_User $user
+ * @var array   $contributor_posts
+ * @var WP_Post $contributor_post
+ * @var string  $success_message
+ * @var string  $pledge_url
+ */
+
+?>
+
+<?php if ( is_user_logged_in() ) : ?>
+
+	<?php echo get_avatar( $user->ID, 96, 'blank', "{$user->login}'s avatar" ); ?>
+
+	<p>Pledged 10 hours a week across two organizations</p> <?php // todo pull from profiles. ?>
+
+	<?php if ( $success_message ) : ?>
+		<div class="notice notice-success notice-alt">
+			<p>
+				<?php echo esc_html( $success_message ); ?>
+			</p>
+		</div>
+	<?php endif; ?>
+
+	<?php if ( $contributor_posts ) : ?>
+		<?php foreach ( $contributor_posts as $contributor_post ) : ?>
+			<?php $pledge = get_post( $contributor_post->post_parent ); ?>
```

Comment:
I have a lot of junk data from testing, but I did end up with a PHP notice because I had a contributor tied to a deleted pledge. Maybe check if the pledge exists here?

Reply:
🤔 We won't ever delete pledges, though, only mark them as `deactivated`, right?

If so, I'm inclined to keep it the way it is. I think notices should always be fixed for expected conditions, but notices for unexpected conditions are intentional, because they let you know that something is wrong.